packages/core/frontend/sirius-components-core/src/onboarding/NewDocumentArea.tsx

```diff
@@ -0,0 +1,206 @@
+    backgroundColor: 'transparent',
+
+    maxWidth: '330px',
+    minWidth: '330px',
+  },
+  cardContent: {
+    height: '100%',
+    maxHeight: '400px',
+    overflowY: 'auto',
+  },
+}));
+
+const createDocumentMutation = gql`
+  mutation createDocument($input: CreateDocumentInput!) {
+    createDocument(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const invokeEditingContextActionMutation = gql`
+  mutation invokeEditingContextAction($input: InvokeEditingContextActionInput!) {
+    invokeEditingContextAction(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+export const NewDocumentArea = ({
+  editingContextId,
+  stereotypeDescriptions,
+  editingContextActions,
+  setSelection,
+  readOnly,
+}: NewDocumentAreaProps) => {
+  const classes = useNewDocumentAreaStyles();
+  const [message, setMessage] = useState('');
+  const [open, setOpen] = useState(false);
+
+  // Document creation
+  const [createDocument, { loading, data, error }] = useMutation(createDocumentMutation);
+  const onCreateDocument = (stereotypeDescriptionId: string) => {
+    const selected = stereotypeDescriptions.find((candidate) => candidate.id === stereotypeDescriptionId);
+    const input: GQLCreateDocumentMutationInput = {
+      id: uuid(),
+      editingContextId,
+      name: selected.documentName,
+      stereotypeDescriptionId,
+    };
+    const variables: GQLCreateDocumentMutationVariables = {
+      input,
+    };
+    createDocument({ variables });
+  };
+  useEffect(() => {
+    if (!loading && !error && data?.createDocument) {
+      const { createDocument } = data;
+      if (createDocument.__typename === 'ErrorPayload') {
+        setOpen(true);
+        setMessage(createDocument.message);
+      }
+    }
+  }, [loading, data, error, setSelection]);
+
+  // EditingContext Action invocation
+  const [
+    invokeEditingContextAction,
+    { loading: loadingEditingContextAction, data: dataEditingContextAction, error: errorEditingContextAction },
+  ] = useMutation(invokeEditingContextActionMutation);
+  const onInvokeEditingContextAction = (actionId: string) => {
+    const input: GQLInvokeEditingContextActionInput = {
+      id: uuid(),
+      editingContextId,
+      actionId,
+    };
+    const variables: GQLInvokeEditingContextActionVariables = {
+      input,
+    };
+    invokeEditingContextAction({ variables });
+  };
+  useEffect(() => {
+    if (
+      !loadingEditingContextAction &&
+      !errorEditingContextAction &&
+      dataEditingContextAction?.invokeEditingContextAction
+    ) {
+      const { invokeEditingContextAction } = dataEditingContextAction;
+      if (invokeEditingContextAction.__typename === 'ErrorPayload') {
+        setOpen(true);
+        setMessage(invokeEditingContextAction.message);
+      }
+    }
+  }, [loadingEditingContextAction, dataEditingContextAction, errorEditingContextAction, setSelection]);
```

Comment:
- Please consider potential errors using the variable `errorEditingContextAction`
- Why are you depending on setSelection here?